src/components/AddCard/index.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { v4 as uuid } from 'uuid';
+import { CAR_COLORS, CAR_PAYMENT_TYPE, SUPPORTED_FILE_TYPES } from '../../constants';
+import CardList from '../CardList';
+import { Car } from '../../models';
+import { CarFormValues } from './models';
+
+import './addCard.css';
+
+const AddCard: React.FunctionComponent = () => {
+  const [cards, setCards] = React.useState<Car[]>([]);
+  const [isSaved, setSaved] = React.useState(false);
+
+  React.useEffect(() => {
+    isSaved && setTimeout(() => setSaved(false), 2000);
+  }, [isSaved]);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<CarFormValues>({ mode: 'onSubmit', reValidateMode: 'onSubmit' });
+  const onSubmit: SubmitHandler<CarFormValues> = (data) => {
+    const { model, brand, productionDate, color, isBooked, paymentType, preview } = data;
+    const files = preview;
+    const file = files && files.length > 0 ? files[0] : null;
+
+    const carCard: Car = {
+      id: uuid(),
+      model,
+      brand,
+      productionDate,
+      color,
+      isBooked,
+      paymentType,
+      src: file ? URL.createObjectURL(file) : '',
+    };
+
+    setCards([...cards, carCard]);
+    setSaved(true);
+    reset();
+  };
+
+  const isUnsupportedFileTypes = (files: FileList): boolean => {
+    const unsupportedFiles = files
+      ? Object.values(files).filter((file) => {
+          return !SUPPORTED_FILE_TYPES.includes(file.type);
+        })
+      : [];
+
+    return unsupportedFiles.length === 0;
+  };
+
+  return (
+    <>
+      {isSaved && <span className="confirmation-message">The data has been saved</span>}
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div className="field">
+          <div>Model</div>
+          <input
+            type="text"
+            data-testid="model-input"
+            {...register('model', { required: 'Fill in Model' })}
+          />
+          {errors.model && <span className="validationError">{errors.model.message}</span>}
```

Comment:
A lot of redundant logic inside render(). Can be at very least extracted into component. Better yet a Hoc that will render together title, input, error message and get error messages from react hook form error context.